src/coreclr/jit/gentree.cpp

```diff
@@ -20059,6 +20059,24 @@ GenTree* Compiler::gtNewSimdCmpOpNode(genTreeOps  op,
 
             if (intrinsic == NI_Illegal)
             {
+                // If we don't have an intrinsic set for this, try "Min(op1, op2) == op1"
+                if (((simdSize == 32) && compOpportunisticallyDependsOn(InstructionSet_AVX2)) ||
+                    ((simdSize == 16) && compOpportunisticallyDependsOn(InstructionSet_SSE41)))
+                {
+                    // We can't use this trick for longs
+                    if (!varTypeIsLong(simdBaseType))
```

Comment:
We can with AVX-512. Might be worth a tracking issue since we're doing that work this release.